src/DDTrace/Integrations/CakePHP/CakePHPIntegration.php

```diff
@@ -71,15 +69,13 @@ function (SpanData $span, array $args) use ($integration) {
                         return;
                     }
 
-                    $integration->rootSpan->setTag(
-                        Tag::RESOURCE_NAME,
-                        $_SERVER['REQUEST_METHOD'] . ' ' . $this->name . 'Controller@' . $request->params['action']
-                    );
-                    $integration->rootSpan->setTag(Tag::HTTP_URL, Router::url($request->here, true));
-                    $integration->rootSpan->setTag('cakephp.route.controller', $request->params['controller']);
-                    $integration->rootSpan->setTag('cakephp.route.action', $request->params['action']);
+                    $integration->rootSpan->resource =
+                        $_SERVER['REQUEST_METHOD'] . ' ' . $this->name . 'Controller@' . $request->params['action'];
+                    $integration->rootSpan->meta[Tag::HTTP_URL] = Router::url($request->here, true);
```

Comment:
`setTag()` does an `Urls::sanitize()` here (i.e. dropping a query string). If I read the CakePHP code correctly, the query string is retained in `Router::url()`.

Reply:
Also verify all other occurrences of Tag::HTTP_URL please.

Reply:
very good catch!